Add variant for unified memory in CoreNEURON

iomaganaris opened the pull request.

iomaganaris committed:
Add variant for unified memory in coreneuron

olupton reviewed (dismissed):
The diff looks good to me, and if we merge this then the GitLab CI on the `master` of CoreNEURON will pick up the change to the recipe.

**However**, the Jenkins CI is no longer running on this `develop` branch and the 2022 deployment is in the `merge-upstream` branch of this repository, so we might prefer to target that and enable this test in https://github.com/BlueBrain/CoreNeuron/pull/738 ? Hopefully this will all come together in the next few days.

iomaganaris commented:
> The diff looks good to me, and if we merge this then the GitLab CI on the `master` of CoreNEURON will pick up the change to the recipe.
> 
> **However**, the Jenkins CI is no longer running on this `develop` branch and the 2022 deployment is in the `merge-upstream` branch of this repository, so we might prefer to target that and enable this test in [BlueBrain/CoreNeuron#738](https://github.com/BlueBrain/CoreNeuron/pull/738) ? Hopefully this will all come together in the next few days.

Okay, makes sense. I've rebased and changed the target branch
